flink-runtime/src/main/java/org/apache/flink/runtime/rest/messages/taskmanager/TaskManagerDetailsInfo.java

```diff
@@ -70,31 +80,39 @@ public TaskManagerDetailsInfo(
                 memoryConfiguration);
 
         this.taskManagerMetrics = Preconditions.checkNotNull(taskManagerMetrics);
+        this.allocatedSlots = Preconditions.checkNotNull(allocatedSlots);
     }
 
     public TaskManagerDetailsInfo(
-            TaskManagerInfo taskManagerInfo, TaskManagerMetricsInfo taskManagerMetrics) {
+            TaskManagerInfoWithSlots taskManagerInfoWithSlots,
+            TaskManagerMetricsInfo taskManagerMetrics) {
         this(
-                taskManagerInfo.getResourceId(),
-                taskManagerInfo.getAddress(),
-                taskManagerInfo.getDataPort(),
-                taskManagerInfo.getJmxPort(),
-                taskManagerInfo.getLastHeartbeat(),
-                taskManagerInfo.getNumberSlots(),
-                taskManagerInfo.getNumberAvailableSlots(),
-                taskManagerInfo.getTotalResource(),
-                taskManagerInfo.getFreeResource(),
-                taskManagerInfo.getHardwareDescription(),
-                taskManagerInfo.getMemoryConfiguration(),
+                taskManagerInfoWithSlots.getTaskManagerInfo().getResourceId(),
+                taskManagerInfoWithSlots.getTaskManagerInfo().getAddress(),
+                taskManagerInfoWithSlots.getTaskManagerInfo().getDataPort(),
+                taskManagerInfoWithSlots.getTaskManagerInfo().getJmxPort(),
+                taskManagerInfoWithSlots.getTaskManagerInfo().getLastHeartbeat(),
+                taskManagerInfoWithSlots.getTaskManagerInfo().getNumberSlots(),
+                taskManagerInfoWithSlots.getTaskManagerInfo().getNumberAvailableSlots(),
+                taskManagerInfoWithSlots.getTaskManagerInfo().getTotalResource(),
+                taskManagerInfoWithSlots.getTaskManagerInfo().getFreeResource(),
+                taskManagerInfoWithSlots.getTaskManagerInfo().getHardwareDescription(),
+                taskManagerInfoWithSlots.getTaskManagerInfo().getMemoryConfiguration(),
+                taskManagerInfoWithSlots.getAllocatedSlots(),
                 taskManagerMetrics);
     }
 
-    @JsonProperty(FIELD_NAME_METRICS)
+    @JsonIgnore
```

Comment:
This is a bit unrelated. We may consider to have this in a hotfix commit, together with properties in `TaskManagerInfo`.